### ISSUE TITLE
Copy/Paste from select mode causes crash

### ISSUE THREAD

**Issue by josephgerth** [andmarti1424/sc-im#698]
I was trying to copy and paste some cells within a spreadsheet. I selected the range AF2-AF11, and was going to paste that into C2-C11. Copying and cutting operate as expected.

When I go to paste the program crashes. I get the error 
`corrupted size vs. previous size zsh: IOT instruction (core dumped) sc-im FILENAME`
whether I had used copy or paste.

**Comment by andmarti1424** [comment]
Could you please share your spreadsheet so I can check it out?
Can you reproduce it from a fresh start after loading the file?
Thanks.

**Comment by andmarti1424** [comment]
@josephgerth, can you share your spreadsheet? Thanks.

**Comment by ei14** [comment]
@andmarti1424 

I reproduced a similar bug. sc-im appears to crash when putting the contents of a visual-mode yank when doing so would update the result of another cell.

Upon opening sc-im, I typed the following:

`=@sum(A1:A1)<Enter>l=1<Enter>vlyhjp`

I use `fish`, so my error message is different:

`fish: Job 1, 'sc-im' terminated by signal SIGSEGV (Address boundary error)`

**Comment by andmarti1424** [comment]
I am afraid I am not able to reproduce the error @ei14.
For instance with this file:

```
newsheet "Sheet1"
movetosheet "Sheet1"
offscr_sc_cols 0
offscr_sc_rows 0
nb_frozen_rows 0
nb_frozen_cols 0
nb_frozen_screenrows 0
nb_frozen_screencols 0
let A0 = 2
let B0 = 1
let A1 = 3
let B1 = 5
goto B1
```
after opening it, if you press `vkhyhjp`... does it crash as well?

**Comment by ei14** [comment]
@andmarti1424

I got a syntax error when opening the file. After pressing escape precisely 8 times, the file opens.

Pressing `vkhyhjp` does not cause a crash with the file you provided.

Since you couldn't reproduce my error, I'll get specific about my setup.

* Latitude E6330 01
* Intel i7-3520M (4) @ 3.600GHz
* Arch Linux 5.16.10-arch1-1
* dwm
* sc-im version 0.8.2

I reproduced my issue in 2 terminal emulators (st and xterm) and in 2 shells (fish and bash).

It seems to never occur if there are no cells that are calculated dependently upon other cells. It is specifically triggered when putting a yanked cell would update the calculation of another cell, but the selection in visual mode did not originate from the cell containing the number that causes the update in the other cell.

Here's something new I discovered: the glitch happens differently depending on whether the sheet open was opened from a file or created in the same session. For example, take this file:

```
let A0 = @sum(A1:A1)
let B0 = @sum(B1:B1)
let C0 = 1
let D0 = 1
goto D0
```

Opening the file and performing `vhyhhhjp` does *not* cause a crash, although it incorrectly populates cell A1 with a bright red "REF" instead of 1.00. Quitting with `:q!` does cause sc-im to exit with the exact same error message as the glitch in question.

However, opening a fresh session of sc-im, recreating the same file manually, and performing the same sequence *does* cause the crash. I.e., the following sequence causes the crash: `=@sum(A1:A1)<Enter>l=@sum(B1:B1)<Enter>l=1<Enter>l=1<Enter>vhyhhhjp`
